addons/mrp/models/stock_move.py

```diff
@@ -157,6 +157,20 @@ def default_get(self, fields_list):
                 defaults['additional'] = True
         return defaults
 
+    @api.model
+    def create(self, values):
+        if values.get('raw_material_production_id'):
+            mo = self.env['mrp.production'].browse(values['raw_material_production_id'])
+            values.update({
+                'group_id': mo.procurement_group_id.id,
+                'reference': mo.name
+            })
+        res = super().create(values)
+        if res.raw_material_production_id and res.raw_material_production_id.state == 'confirmed' and not res.bom_line_id:
+            res._adjust_procure_method()
+            res._action_confirm()
```

Comment:
actually.... we already have an "additional" field on the move, with an "autoconfirm" mechanism working on those moves
maybe you could use the same concept here?